src/main/java/nextstep/qna/domain/Answer.java

```diff
@@ -29,51 +18,51 @@ public Answer(NsUser writer, Question question, String contents) {
     }
 
     public Answer(Long id, NsUser writer, Question question, String contents) {
-        this.id = id;
-        if(writer == null) {
+        this(id, question, new PostContent(writer,contents));
+    }
+
+    public Answer(Long id, Question question, PostContent postContent) {
+        super(id);
+        if(postContent.isWriter()) {
             throw new UnAuthorizedException();
         }
 
         if(question == null) {
             throw new NotFoundException();
         }
 
-        this.writer = writer;
         this.question = question;
-        this.contents = contents;
+        this.postContent = postContent;
     }
 
-    public Long getId() {
-        return id;
-    }
-
-    public Answer setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
-    public boolean isDeleted() {
-        return deleted;
+    public void markAsDeleted(NsUser loginUser) throws CannotDeleteException {
+        validateDeletableBy(loginUser);
+        delete();
     }
 
     public boolean isOwner(NsUser writer) {
-        return this.writer.equals(writer);
+        return this.postContent.isOwner(writer);
     }
 
     public NsUser getWriter() {
-        return writer;
-    }
-
-    public String getContents() {
-        return contents;
+        return this.postContent.getWriter();
     }
 
     public void toQuestion(Question question) {
         this.question = question;
     }
 
+    public void validateDeletableBy(NsUser loginUser) throws CannotDeleteException {
```

Comment:
```suggestion
    private void validateDeletableBy(NsUser loginUser) throws CannotDeleteException {
```
private으로 구현해도 되지 않을까?